### PR TITLE
Sensor reading

### PR DESCRIPTION
**Pull request by jamiesanson**
This pull request covers sensor reading from the micro version of the MPU9250 sensor module. The library was written to make the Arduino/Teensy sketch code less messy.

### CODE REVIEW

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+  _mscale = MScale;
+  _mmode = Mmode;
+}
+
+void MPU9250_helper::initAK8963(float * destination)
+{
+  // First extract the factory calibration for each magnetometer axis
+  uint8_t rawData[3];  // x/y/z gyro calibration data stored here
+  writeByte(AK8963_ADDRESS, AK8963_CNTL, 0x00); // Power down magnetometer
+  delay(10);
```
Comment (marcelvanworkum) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp:17]: Not sure of the convention for C, but I would usually make the 10 a constant if it's gonna be used in multiple places.

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+  writeByte(AK8963_ADDRESS, AK8963_CNTL, 0x0F); // Enter Fuse ROM access mode
+  delay(10);
+  readBytes(AK8963_ADDRESS, AK8963_ASAX, 3, &rawData[0]);  // Read the x-, y-, and z-axis calibration values
+  destination[0] =  (float)(rawData[0] - 128)/256. + 1.;   // Return x-axis sensitivity adjustment values, etc.
```
Comment (marcelvanworkum) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp]: Again, not sure on convention but 128 and 256 could be constants.

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+  destination[1] =  (float)(rawData[1] - 128)/256. + 1.;
```
Comment (marcelvanworkum) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp]: And what is the '+ 1.;' Is the period making it a floating point? If so might be clearer to just be 1.0?

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+  }
+
+  accel_bias[0] /= (int32_t) packet_count; // Normalize sums to get average count biases
+  accel_bias[1] /= (int32_t) packet_count;
+  accel_bias[2] /= (int32_t) packet_count;
+  gyro_bias[0]  /= (int32_t) packet_count;
+  gyro_bias[1]  /= (int32_t) packet_count;
+  gyro_bias[2]  /= (int32_t) packet_count;
+
+  if(accel_bias[2] > 0L) {accel_bias[2] -= (int32_t) accelsensitivity;}  // Remove gravity from the z-axis accelerometer bias calculation
```
Comment (marcelvanworkum) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp]: Not sure how I feel about one line if statements

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+          return 2.0/32768.0;
+    case AFS_4G:
+          return 4.0/32768.0;
+    case AFS_8G:
+          return 8.0/32768.0;
+    case AFS_16G:
+          return 16.0/32768.0;
+  }
+
+  return 0;
```
Comment (marcelvanworkum) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp]: Should this return 0 not be the 'default' case of the switch? Does C even have that?

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+          return 250.0/32768.0;
+    case GFS_500DPS:
+          return 500.0/32768.0;
+    case GFS_1000DPS:
+          return 1000.0/32768.0;
+    case GFS_2000DPS:
+          return 2000.0/32768.0;
+  }
+
+  return 0;
```
Comment (marcelvanworkum) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp]: same thing

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+  {
+    // Possible magnetometer scales (and their register bit settings) are:
+    // 14 bit resolution (0) and 16 bit resolution (1)
+    case MFS_14BITS:
+          return 10.*4912./8190.;
+    case MFS_16BITS:
+          return 10.*4912./32760.0;
+  }
+
+  return 0;
```
Comment (marcelvanworkum) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp]: ^

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp (hunk 1 above)
Comment (jamiesanson) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp:17]: @Recursively With regards to Arduino code I don't think these need to be, as delays are used all over the place with arbitrary counts as parameters

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.h [hunk 8]
```diff
@@ -0,0 +1,215 @@
+/*
```
Comment (marcelvanworkum) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.h:1]: Should the commented out defines not be removed?

File: test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp (hunk 3 above)
Comment (jamiesanson) [inline test_flight_basic_logging/libraries/MPU9250_helper/MPU9250_helper.cpp]: @Recursively Yeah it does. I'll go ahead and change all the occurances of that

**Pull request merged**